src/ngraph/descriptor/tensor.cpp

```diff
@@ -22,20 +22,43 @@ using namespace ngraph;
 using namespace std;
 
 descriptor::Tensor::Tensor(const element::Type& element_type,
-                           const Shape& shape,
+                           const PartialShape& pshape,
                            const std::string& name)
     : m_element_type(element_type)
-    , m_shape(shape)
+    , m_shape(pshape.is_complete() ? pshape.to_shape() : Shape{})
+    , m_partial_shape(pshape)
     , m_name(name)
 {
 }
 
-void descriptor::Tensor::set_tensor_type(const element::Type& element_type, const Shape& shape)
+void descriptor::Tensor::set_tensor_type(const element::Type& element_type,
+                                         const PartialShape& pshape)
```

Comment:
I think it might be better to just set partial shape and have validate do that, and have a pass that switches over to m_shape when shapes are fully known.

Reply:
I agree that maintaining `m_shape` in parallel like this is awkward. The motivation for the moment is that I want to maintain backwards compatibility during the transition with certain APIs that are returning a `const Shape&`.